index.js

```diff
@@ -34,340 +32,72 @@ function start(clusterProcessId = 0) { // eslint-disable-line max-statements
   console.log(`Started node process ${clusterProcessId}`);
 
   /*
-   * Heroku support suggested we use segfault-handler but it's not available
-   * for node 8 yet.
+   * Express app
    */
-
-  // const SegfaultHandler = require('segfault-handler');
-  // SegfaultHandler.registerHandler('crash.log');
-
-  const featureToggles = require('feature-toggles');
-  const logEnvVars = require('./utils/logEnvVars');
-
-  if (conf.newRelicKey) {
-    require('newrelic');
-  }
-
-  const helmet = require('helmet');
-  const swaggerTools = require('swagger-tools');
-
-  const errorHandler = require('./api/v1/errorHandler');
-  const path = require('path');
-  const fs = require('fs');
-  const yaml = require('js-yaml');
-  const ipfilter = require('express-ipfilter');
-  const rejectMultipleXForwardedFor =
-    require('./config/rejectMultipleXForwardedFor');
-  const bodyParser = require('body-parser');
-  const env = conf.environment[conf.nodeEnv];
-  const ENCODING = 'utf8';
-  const compress = require('compression');
-  const cors = require('cors');
-  const etag = require('etag');
-
-  // set up server side socket.io and redis publisher
-  const express = require('express');
-  const enforcesSSL = require('express-enforces-ssl');
-
-  const app = express();
+  const app = require('./express').app;
 
   /*
-   * If clusterProcessId is 0, we're running in single-process mode (i.e.
-   * non-throng), so log the env vars out.
-   * If clusterProcessId > 0, we have multiple throng workers, and in that case
-   * we only want to log out the env vars once, since they're all the same
-   * across all the throng workers, so in this case just do the logging if
-   * clusterProcessId is 1.
+   * Sample store
    */
-  if (clusterProcessId < 2) logEnvVars.log(process.env);
+  require('./cache/sampleStoreInit').init();
 
   /*
-   * Call this *before* the static pages and the API routes so that both the
-   * static pages *and* the API responses are compressed (gzip).
+   * Clock jobs
+   * If the clock dyno is NOT enabled, schedule all the scheduled jobs right
+   * from here.
    */
-  app.use(compress());
+  if (!featureToggles.isFeatureEnabled('enableClockProcess')) {
+    require('./clock/index'); // eslint-disable-line global-require
```

Comment:
Does this even work? If I'm starting up the clock INSIDE a web dyno, won't I have multiple clocks if I have multiple dynos, and multiple processes per dyno?
I think we should no longer support running the clock inside the main web process.

Reply:
This is exactly how it was before.

Yes, it would create multiple clocks, which is why when running in heroku enableClockProcess is set and this line doesn't run.

This was the discussion we had last week - we decided it was important to support a minimum-effort deployment where you can run a single command to start a single process and have everything just work.

The way the environment variable works here is kind of strange, since we're relying on a variable that isn't actually required to run the separate clock process. I did think about trying to detect whether it was running standalone or with a separate clock process so we don't need an environment variable at all, but that didn't seem very promising. I decided it was best to stick with the way it works now.

Reply:
If we wanted to clean this up, we could require that the clock process only runs if the env var is set, so there's no way to run it in both places. And maybe prevent it from running within the web process if there are multiple processes.
That would be a separate story - this PR is specifically for allowing the tests to be run without the clock jobs, it doesn't change anything about how it works when running the full app.

Reply:
Let's get a story out there to *only* run the clock jobs in a separate process.

Reply:
"Let's get a story out there to only run the clock jobs in a separate process."

From my comment above:
"This was the discussion we had last week - we decided it was important to support a minimum-effort deployment where you can run a single command to start a single process and have everything just work."

Did something change? Do we need to discuss again?

Reply:
I changed my mind because our current implementation is flawed because it can allow multiple clocks to be running, which we would never want in a production environment, and because in dev mode a developer can just use `heroku local` (so there is no big huge burden of having to start multiple processes).

Reply:
We could fix that without removing the ability to run in a separate process by tieing the clock process to the environment variable, and when running in the web process doing the same thing we do for env var logging - only run if process id is 0 or 1. 

I will create a story and we can discuss further at backlog grooming.